src/system/pokemon-data.ts

```diff
@@ -140,7 +140,7 @@ export default class PokemonData {
     this.fusionLuck = source.fusionLuck ?? (source.fusionShiny ? source.fusionVariant + 1 : 0);
     this.fusionTeraType = (source.fusionTeraType ?? 0) as PokemonType;
 
-    this.boss = (source instanceof EnemyPokemon && !!source.bossSegments) || (!this.player && !!source.boss);
+    this.boss = (source.isEnemy() && !!source.bossSegments) || (!this.player && !!source.boss);
```

Comment:
```suggestion
    this.boss = (source instanceof EnemyPokemon && !!source.bossSegments) || (!this.player && !!source.boss);
```
You aren't able to remove this yet (at least not this way).